offer/src/main/java/org/shareio/backend/core/usecases/service/SearchOffersUseCaseService.java

```diff
@@ -0,0 +1,98 @@
+package org.shareio.backend.core.usecases.service;
+
+import lombok.AllArgsConstructor;
+import org.shareio.backend.core.model.Offer;
+import org.shareio.backend.core.model.User;
+import org.shareio.backend.core.model.vo.Category;
+import org.shareio.backend.core.model.vo.Condition;
+import org.shareio.backend.core.model.vo.OfferSortType;
+import org.shareio.backend.core.usecases.port.in.GetAllUserIdListUseCaseInterface;
+import org.shareio.backend.core.usecases.port.in.GetAverageUserReviewValueUseCaseInterface;
+import org.shareio.backend.core.usecases.port.in.SearchOffersUseCaseInterface;
+import org.shareio.backend.core.usecases.port.out.GetAllOffersDaoInterface;
+import org.shareio.backend.core.usecases.port.out.GetUserProfileDaoInterface;
+import org.shareio.backend.core.usecases.util.DistanceCalculator;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDate;
+import java.util.*;
+
+@AllArgsConstructor
+@Service
+public class SearchOffersUseCaseService implements SearchOffersUseCaseInterface {
+
+    GetUserProfileDaoInterface getUserProfileDaoInterface;
+    GetAllOffersDaoInterface getAllOffersDaoInterface;
+    GetAllUserIdListUseCaseInterface getAllUserIdListUseCaseInterface;
+    GetAverageUserReviewValueUseCaseInterface getAverageUserReviewValueUseCaseInterface;
+
+    @Override
+    public List<UUID> getOfferListMeetingCriteria(UUID userId, String title, String category, String condition, Double distance, Double score, LocalDate creationDate, String sortType) {
+        User user = getUserProfileDaoInterface.getUserDto(userId).map(User::fromDto).orElseThrow(NoSuchElementException::new);
+        List<Offer> offers = getAllOffersDaoInterface.getAllOffers().stream().map(Offer::fromDto).toList();
+
+        // FILTER
+        if (Objects.nonNull(title) && !title.isBlank()) {
+            offers = offers.stream()
+                    .filter(offer -> offer.getTitle().toUpperCase().contains(title.toUpperCase())).toList();
```

Comment:
Change implementation to the one in getOffersByName

Reply:
What's wrong with this one?

Reply:
When the offer title is "Ładny szop" we don't want it to be found by typing "SZ"